### ISSUE TITLE
TimeSeriesTransformerForPrediction model unused parameters Runtime error in Distributed environment

### ISSUE THREAD

**Issue by maeschbacher** [huggingface/transformers#25620]
### System Info

transformers==4.31
accelerate==0.21

I'm trying to run TimeSeriesTransformerForPrediction in a distributed environment based on the following notebook: https://huggingface.co/blog/time-series-transformers

If I run the notebook as is I get the following error immediately after the first loss is calcualted: 
```
 Expected to have finished reduction in the prior iteration before starting a new one. This error indicates that your module has parameters that were not used in producing loss. You can enable unused parameter detection by passing the keyword argument `find_unused_parameters=True` to `torch.nn.parallel.DistributedDataParallel`, and by 
making sure all `forward` function outputs participate in calculating loss. 
If you already have done the above, then the distributed data parallel module wasn't able to locate the output tensors in the return value of your module's `forward` function. Please include the loss function and the structure of the return value of `forward` of your module when reporting this issue (e.g. list, dict, iterable).
Parameter indices which did not receive grad for rank 0: 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146
 In addition, you can set the environment variable TORCH_DISTRIBUTED_DEBUG to either INFO or DETAIL to print out information about which particular parameters did not receive gradient on this rank as part of this error
```

If I modify the accelerator in the following way:
```
ddp_kwargs = DistributedDataParallelKwargs(find_unused_parameters=True)
accelerator = Accelerator(kwargs_handlers=[ddp_kwargs])
```
It will usually run for many iterations but it randomly fails with an error message similar to above.

The code appears to run fine in cpu mode or on GPU in a non-distributed environment.

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run the following notebook in a distributed GPU environment: https://huggingface.co/blog/time-series-transformers

### Expected behavior

I would expect the model to train and perform `.backward(loss)` without error.

**Comment by ArthurZucker** [comment]
cc @kashif

**Comment by ArthurZucker** [comment]
I don't think @kashif will have time to look at this, so marking as Good Difficult issue.
